src/pages/centers.tsx

```diff
@@ -52,12 +66,93 @@ interface CohortDetails {
   customFields?: CustomField[];
 }
 
+const mapFields = (formFields: any, response: any) => {
+  let initialFormData: any = {};
+  console.log("response", response?.results?.cohortDetails[0]);
+  // Extract the cohort details from the response
+  const cohortDetails = response?.results?.cohortDetails?.[0] || {};
+
+  formFields.fields.forEach((item: any) => {
+    const customFieldValue = cohortDetails?.customFields?.find(
+      (field: any) => field.fieldId === item.fieldId
+    );
+
+    const getValue = (data: any, field: any) => {
+      if (item.default) {
+        return item.default;
+      }
+      if (item?.isMultiSelect) {
+        if (data[item.name] && item?.maxSelections > 1) {
+          return [field?.value];
+        } else if (item?.type === "checkbox") {
+          return String(field?.value).split(",");
+        } else {
+          return field?.value?.toLowerCase();
+        }
+      } else if (item?.type === "radio") {
+        return field?.value || null;
+      } else if (item?.type === "numeric") {
+        return parseInt(String(field?.value));
+      } else if (item?.type === "text") {
+        return String(field?.value);
+      } else {
+        if (field?.value === "FEMALE" || field?.value === "MALE") {
+          return field?.value?.toLowerCase();
+        }
+        return field?.value?.toLowerCase();
+      }
+    };
+
+    if (item.coreField) {
+      if (item?.isMultiSelect) {
```

Comment:
@upendraTekdi move this logic to some common file so that we can use it in any other component